installer/pkg/components/image-builder-mk3/configmap.go

```diff
@@ -23,6 +23,13 @@ import (
 )
 
 func configmap(ctx *common.RenderContext) ([]runtime.Object, error) {
+	var registryName string
+	if pointer.BoolDeref(ctx.Config.ContainerRegistry.InCluster, false) {
+		registryName = fmt.Sprintf("%s.%s", dockerregistry.RegistryName, ctx.Config.Domain)
+	} else {
```

Comment:
```suggestion
	} else if ctx.Config.ContainerRegistry.External != nil {
```

} else { return fmt.Error }